docker-compose down panics with `-p` flag

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.

If you suspect your issue is a bug, please edit your issue description to
include the BUG REPORT INFORMATION shown below. If you fail to provide this
information within 7 days, we cannot debug your issue and will close it. We
will, however, reopen it if you later provide the information.

For more information about reporting issues, see
https://github.com/docker/compose-cli/blob/master/CONTRIBUTING.md#reporting-other-issues

---------------------------------------------------
GENERAL SUPPORT INFORMATION
---------------------------------------------------

The GitHub issue tracker is for bug reports and feature requests.
General support can be found at the following locations:

- Docker Support Forums - https://forums.docker.com
- Docker Community Slack - https://dockr.ly/slack
- Post a question on StackOverflow, using the Docker tag

---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------
Use the commands below to provide key information from your environment:
You do NOT have to include this information if this is a FEATURE REQUEST
-->

**Description**

Running `docker-compose --compatibility -p myproject down -v --rmi all` is panicking with a `nil` pointer: 

When I omit the `-p` flag, it works as expected.

**Steps to reproduce the issue:**
1. `docker-compose --compatibility -p myproject up -d`
2. Wait for project to come up.
3. `docker-compose --compatibility -p myproject down -v --rmi all`

**Describe the results you received:**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1ba08df]

goroutine 28 [running]:
github.com/docker/compose/v2/pkg/compose.(*composeService).getServiceImages(...)
        github.com/docker/compose/v2/pkg/compose/down.go:149
github.com/docker/compose/v2/pkg/compose.(*composeService).ensureImagesDown(0xc000433920, {0x20a01c8?, 0xc00046c0c0}, {0x7ff7bfeff90a, 0x6}, {0x0, 0x0, 0x0, {0x7ff7bfeff91f, 0x3}, ...}, ...)
        github.com/docker/compose/v2/pkg/compose/down.go:119 +0xff
github.com/docker/compose/v2/pkg/compose.(*composeService).down(0xc000433920, {0x20a01c8, 0xc00046c0c0}, {0x7ff7bfeff90a, 0x6}, {0x0, 0x0, 0x0, {0x7ff7bfeff91f, 0x3}, ...})
        github.com/docker/compose/v2/pkg/compose/down.go:85 +0x505
github.com/docker/compose/v2/pkg/compose.(*composeService).Down.func1({0x20a01c8, 0xc00046c0c0})
        github.com/docker/compose/v2/pkg/compose/down.go:39 +0x85
github.com/docker/compose/v2/pkg/progress.Run.func1({0x20a01c8?, 0xc00046c0c0?})
        github.com/docker/compose/v2/pkg/progress/writer.go:61 +0x27
github.com/docker/compose/v2/pkg/progress.RunWithStatus.func2()
        github.com/docker/compose/v2/pkg/progress/writer.go:82 +0x87
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
        golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x8d
```


**Describe the results you expected:**

The project will gracefully shut down.


**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker compose version`:**

```
Docker Compose version 2.4.0
```

**Output of `docker info`:**

```
Client:
 Context:    colima
 Debug Mode: false

Server:
 Containers: 2
  Running: 0
  Paused: 0
  Stopped: 2
 Images: 6
 Server Version: 20.10.11
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 1e5ef943eb76627a6d3b6de8cd1ef6537f393a71
 runc version: b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 init version: 
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.93-0-virt
 Operating System: Alpine Linux v3.14
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.942GiB
 Name: colima
 ID: OLG7:Q7OL:PIVA:VZIE:VELJ:XMC4:ZGMU:EJDE:HFYM:Z3TF:IPDR:2KPJ
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

```

**Additional environment details:**
Host: macOS Monterey 12.3
Using colima version 0.3.4:
```
colima version 0.3.4
git commit: 5a4a70481ca8d1e794677f22524e3c1b79a9b4ae

runtime: docker
arch: x86_64
client: v20.10.12
server: v20.10.11
```

--- a/pkg/compose/down.go
+++ b/pkg/compose/down.go
@@ -82,7 +82,7 @@ func (s *composeService) down(ctx context.Context, projectName string, options a
 	ops := s.ensureNetworksDown(ctx, project, w)
 
 	if options.Images != "" {
-		ops = append(ops, s.ensureImagesDown(ctx, projectName, options, w)...)
+		ops = append(ops, s.ensureImagesDown(ctx, project, options, w)...)
 	}
 
 	if options.Volumes {
@@ -114,9 +114,9 @@ func (s *composeService) ensureVolumesDown(ctx context.Context, project *types.P
 	return ops
 }
 
-func (s *composeService) ensureImagesDown(ctx context.Context, projectName string, options api.DownOptions, w progress.Writer) []downOp {
+func (s *composeService) ensureImagesDown(ctx context.Context, project *types.Project, options api.DownOptions, w progress.Writer) []downOp {
 	var ops []downOp
-	for image := range s.getServiceImages(options, projectName) {
+	for image := range s.getServiceImages(options, project) {
 		image := image
 		ops = append(ops, func() error {
 			return s.removeImage(ctx, image, w)
@@ -144,15 +144,15 @@ func (s *composeService) ensureNetworksDown(ctx context.Context, project *types.
 	return ops
 }
 
-func (s *composeService) getServiceImages(options api.DownOptions, projectName string) map[string]struct{} {
+func (s *composeService) getServiceImages(options api.DownOptions, project *types.Project) map[string]struct{} {
 	images := map[string]struct{}{}
-	for _, service := range options.Project.Services {
+	for _, service := range project.Services {
 		image := service.Image
 		if options.Images == "local" && image != "" {
 			continue
 		}
 		if image == "" {
-			image = getImageName(service, projectName)
+			image = getImageName(service, project.Name)
 		}
 		images[image] = struct{}{}
 	}

